source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,351 @@ Documents
-- Table comparison of classes (columns for: implements Map, provides type safety, is an interface or class, compatibility with server versions
-- Recommend Document for general use and why.
-- Duplicate field name admonition (I forgot what this is)
+A MongoDB **document** is a data structure that contains key/value pairs in
+binary JSON (BSON) format and is the base unit in the MongoDB data hierarchy.
+
+Each **document** belongs to a single **collection** and each **collection**
+belongs to a single **database**. For more information, see our MongoDB manual
+entry on :manual:`Documents </core/document>`.
+
+The Java driver and BSON library include the following document classes that
+help you access and manipulate the BSON data in documents:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 10 20 20
+
+   * - Name
+     - Package
+     - Implements Map
+     - Recommended Usage
+
+   * - ``Document``
+     - ``org.bson``
+     - Yes, implements ``Map<String, Object>``
+     - When you want a flexible and concise data representation.
+
+   * - ``BsonDocument``
+     - ``org.bson``
+     - Yes, implements ``Map<String, BsonValue>``
+     - When you need a type-safe API.
+
+   * - ``BasicDBObject``
+     - ``com.mongodb``
+     - No
+     - When you are migrating your application from a legacy driver version.
+
+While you can use any of the document classes in your application, we
+recommend that you use the ``Document`` class since it can concisely
+represent dynamically structured documents of any complexity. It
+implements the ``Map<String, Object>`` interface which enables it to use
+loosely-typed values.
+
+.. warning:: Duplicate Field Names
+
+   Although a document is valid if it contains two duplicate keys, the driver
+   only sets the value of the key to the value in the last instance.
 
 Document
 --------
 
-- Description
-- Chart that shows mapping between BSON and Java types (similar to the one here https://mongodb.github.io/mongo-java-driver/4.1/bson/documents/)
-- Code example of declaring one and appending fields
-- Code example of get() for various types
-- Link to Fundamentals > CRUD for more information
-- Link to API documentation
+The ``Document`` class offers a flexible representation of a BSON document.
+You can access and manipulate document data using Java types with this
+class. See the following table for mappings between frequently-used BSON
+and Java types:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - BSON type
+     - Java type
+   * - Array
+     - ``java.util.List``
+   * - Binary
+     - ``org.bson.types.Binary``
+   * - Boolean
+     - ``java.lang.Boolean``
+   * - Date
+     - ``java.util.Date``
+   * - Document
+     - ``org.bson.Document``
+   * - Double
+     - ``java.lang.Double``
+   * - Int32
+     - ``java.lang.Integer``
+   * - Int64
+     - ``java.lang.Long``
+   * - Null
+     - ``null``
+   * - ObjectId``
```

Comment:
```suggestion
   * - ObjectId
```
I think you accidentally a monospace!

Reply:
Whoops, good catch. Even the autobuilder did not warn me about that.